receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -176,6 +179,18 @@ func (c *kafkaTracesConsumer) Start(_ context.Context, host component.Host) erro
 			return err
 		}
 	}
+	// takes precedence over format type and overwrites the unmarshaler
+	if c.config.Encoding != nil {
+		encoding := host.GetExtensions()[*c.config.Encoding]
+		if encoding == nil {
+			return fmt.Errorf("unknown encoding %q", c.config.Encoding)
+		}
+		unmarshaler, _ := encoding.(ptrace.Unmarshaler)
```

Comment:
It is probably worth validating it cast correctly here since we are taking a dependancy on an external type.